lib/improver/wind_downscaling.py

```diff
@@ -0,0 +1,967 @@
+        rcmask = np.copy(hcmask)
+        if not self.l_no_winddownscale:
+            rcmask[self.z_0 <= 0] = False
+            rcmask[np.isnan(self.z_0)] = False
+        return hcmask, rcmask
+
+    @staticmethod
+    def sigma2hover2(sigma):
+        """Calculate the half peak-to-trough height.
+
+        The ancillary data used to estimate the peak to trough height
+        contains the standard deviation of height in a cell. For
+        sine-waves, this relates to the amplitude of the wave as:
+
+        Amplitude = sigma * sqrt(2)
+
+        The amplitude would correspond to half the peak-to-trough
+        height (h_o_2).
+
+        Parameters:
+        -----------
+            sigma: 2D array
+                standard deviation of height in grid cell.
+
+        Returns:
+        --------
+            h_o_2: 2D array
+                of half peak-to-trough height.
+
+        Comments:
+            Points that had sigma = 0 (i.e. sea points) are set to
+            RMDI.
+
+        """
+        h_o_2 = np.ones(sigma.shape) * RMDI
+        h_o_2[sigma > 0] = sigma[sigma > 0] * np.sqrt(2.0)
+        return h_o_2
+
+    def calc_wav(self):
+        """Calculate wavenumber k of typical orographic lengthscale.
+
+        Function to calculate wavenumber k of typical orographic
+        lengthscale L:
+            k = 2*pi / L (1)
+
+        L is approximated from half the peak-to-trough height h_over_2
+        and the silhoutte roughness a_over_s (average of up-slopes per
+        unit length over several cross-sections through a grid cell)
+        as:
+            L = 2*h_over_2 / a_over_s (2)
+
+        a_over_s is dimensionless since it is the sum of up-slopes
+        measured in the same unit lengths as it is calculated over.
+
+        h_over_2 is calculated from the standard deviation of height in
+        a grid cell, sigma, as:
+            h_over_2 = sqrt(2) * sigma
+
+        which is based on the assumptions of sine waves, see
+        sigma2hover2.
+
+        From eq. (1) and (2) it follows that:
+            k = 2*pi / (2*h_over_2 / a_over_s)
+              = a_over_s * pi / h_over_2
+
+        Returns:
+        --------
+        wavn: 2D np.array
+            wavenumber in units of inverse units of supplied h_over_2.
+
+        """
+        wavn = np.ones(self.a_over_s.shape) * RMDI
+        wavn[self.hcmask] = (self.a_over_s[self.hcmask] /
+                             self.h_over_2[self.hcmask]*np.pi)
+        wavn[wavn > np.pi/self.dx_min] = np.pi/self.dx_min
+        wavn[self.h_over_2 == 0] = RMDI
+        wavn[abs(wavn) < np.pi/self.dx_max] = np.pi/self.dx_max
+        return wavn
+
+    def calc_h_ref(self):
+        """Calculate the reference height for roughness correction.
+
+        The reference height below which the flow is in equilibrium
+        with the vegetative roughness is proportional to 1/wavenum
+        (Howard & Clark, 2007).
+
+        Vosper (2009) and Clark (2009) argue that at the reference
```

Comment:
OK, I'll just tag @bjwheltor here to give some thought to the use of internal documents within the IMPROVER code.